### ISSUE TITLE
aws_ami owner should be optional if you have a specific AMI ID

### ISSUE THREAD

**Issue by wrschneider** [hashicorp/terraform-provider-aws#14158]
It should work the same way as the AWS CLI, where you can do something like 

```
aws ec2 describe-images --image-ids ami-xxxx
```

It's annoying to have to add the owner when you already know the specific image ID.

**Comment by bflad** [comment]
Hi @wrschneider 👋  Thank you for filing this. Please note that this requirement is due to the publicly reported [CVE-2018-15869](https://nvd.nist.gov/vuln/detail/CVE-2018-15869), so we may be hesitant to make this optional again for security risks.

**Comment by wrschneider** [comment]
Thanks for the context - I didn't realize there were such a good reason for it.

Is that CVE an issue even if you have a specific AMI ID?   I agree it makes sense if you were filtering on some other attributes that don't uniquely determine an AMI.

**Comment by bflad** [comment]
[AMI IDs are unique for the AWS region](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html) you're working in. Theoretically, yes, we could potentially replace the simple [`Required: true` schema declaration](https://github.com/terraform-providers/terraform-provider-aws/blob/2fb5645ae203859cd5cad52bf0f0c88823b721ad/aws/data_source_aws_ami.go#L45-L53) we use now with a logic based approach (via `CustomizeDiff` to ensure it continues to validate the CVE issue during plan-time), but it will introduce some new complexity into the resource.

**Comment by tculp** [comment]
Just ran into this issue as well. Since the aws CLI has the owner field as optional, I don't understand why it shouldn't be optional for this provider. Having an option for "any" or "*", as mentioned above, would also satisfy our needs.

**Comment by mpagot** [comment]
This request is very important for some images. Here an example using deprecated images:

Look for an image using its AMI:

```
$ aws ec2 describe-images --region us-west-2 --image-ids ami-0aeb911096b16301d --query 'Images[][Name,ImageOwnerAlias,
OwnerId,ImageId,DeprecationTime]'
[
    [
        "suse-sles-sap-15-sp1-byos-v20221110-hvm-ssd-x86_64-6066beac-a43b-4ad0-b5fe-f503025e4747",
        "aws-marketplace",
        "679593333241",
        "ami-0aeb911096b16301d",
        "2023-06-19T22:50:00.000Z"
    ]
]
```

Even the `aws cli` is not able to find it if searched by name:

```
% aws ec2 describe-images --region us-west-2  --filters '[{"Name": "name", "Values": ["suse-sles-sap-15-sp1-byos-v2022
1110*"]}]' --query 'Images[][Name,ImageOwnerAlias,OwnerId,ImageId,State,DeprecationTime]'

[]
```

It is to say that also the `aws cli` has strongly different behavior when using `--image-ids`.

So code like https://github.com/SUSE/qe-sap-deployment/tree/main/terraform/aws/modules/get_os_image has no chance to work using `ami-id`.

**Comment by iancward** [comment]
This was apparently implemented in version [4.21.0](https://github.com/hashicorp/terraform-provider-aws/blob/release/4.x/CHANGELOG.md#4210-june-30-2022).

**Comment by DrFaust92** [comment]
closed via #25566
